File: shell/ejecutar_script.py
#!/usr/bin/env python3

import os
import sys
import subprocess
from pathlib import Path

def main():
    print("🚀 Ejecutando script desde Python...")
    print()

    # Obtener el directorio actual
    current_dir = Path(__file__).parent.resolve()

    # Ruta al script
    script_path = current_dir / "script_ejemplo.sh"

    # Verificar que el script existe
    if not script_path.exists():
        print(f"❌ Error: el script no existe en {script_path}", file=sys.stderr)
        sys.exit(1)

    # Parámetros para el script
    nombre = "María"
    edad = "30"

    # Si se pasan argumentos desde la línea de comandos, usarlos
    if len(sys.argv) > 1:
        nombre = sys.argv[1]
    if len(sys.argv) > 2:
        edad = sys.argv[2]

    print(f"📋 Ejecutando: {script_path} {nombre} {edad}")
    print("─────────────────────────────────────")

    try:
        # Ejecutar el script y capturar la salida
        result = subprocess.run(
            [str(script_path), nombre, edad],
            capture_output=True,
            text=True,
            check=False
        )

        # Mostrar stdout
        if result.stdout:
            print(result.stdout, end='')

        # Mostrar stderr si existe
        if result.stderr:
            print(f"⚠️  Stderr:\n{result.stderr}", file=sys.stderr, end='')

        # Verificar el código de retorno
        if result.returncode != 0:
            print(f"\n❌ El script terminó con código de error: {result.returncode}", file=sys.stderr)
            sys.exit(result.returncode)

        print()
        print("✅ Script ejecutado exitosamente desde Python")

    except FileNotFoundError:
        print(f"❌ Error: no se pudo encontrar el script en {script_path}", file=sys.stderr)
        sys.exit(1)
    except PermissionError:
        print(f"❌ Error: no hay permisos para ejecutar {script_path}", file=sys.stderr)
        print("Intenta ejecutar: chmod +x script_ejemplo.sh", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error inesperado: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()

File: shell/ejecutar_script.go
package main

import (
	"bytes"
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
)

func main() {
	fmt.Println("🚀 Ejecutando script desde Go...")
	fmt.Println()

	// Obtener el directorio actual
	dir, err := os.Getwd()
	if err != nil {
		fmt.Fprintf(os.Stderr, "❌ Error obteniendo directorio: %v\n", err)
		os.Exit(1)
	}

	// Ruta al script
	scriptPath := filepath.Join(dir, "script_ejemplo.sh")

	// Verificar que el script existe
	if _, err := os.Stat(scriptPath); os.IsNotExist(err) {
		fmt.Fprintf(os.Stderr, "❌ Error: el script no existe en %s\n", scriptPath)
		os.Exit(1)
	}

	// Parámetros para el script
	nombre := "Juan"
	edad := "25"

	// Si se pasan argumentos desde la línea de comandos, usarlos
	if len(os.Args) > 1 {
		nombre = os.Args[1]
	}
	if len(os.Args) > 2 {
		edad = os.Args[2]
	}

	fmt.Printf("📋 Ejecutando: %s %s %s\n", scriptPath, nombre, edad)
	fmt.Println("─────────────────────────────────────")

	// Crear el comando
	cmd := exec.Command(scriptPath, nombre, edad)

	// Buffers para capturar stdout y stderr
	var stdout, stderr bytes.Buffer
	cmd.Stdout = &stdout
	cmd.Stderr = &stderr

	// Ejecutar el comando
	err = cmd.Run()

	// Mostrar la salida
	if stdout.Len() > 0 {
		fmt.Print(stdout.String())
	}

	if stderr.Len() > 0 {
		fmt.Fprintf(os.Stderr, "⚠️  Stderr:\n%s", stderr.String())
	}

	// Verificar si hubo error
	if err != nil {
		fmt.Fprintf(os.Stderr, "\n❌ Error ejecutando script: %v\n", err)
		os.Exit(1)
	}

	fmt.Println()
	fmt.Println("✅ Script ejecutado exitosamente desde Go")
}
